tests/test_reportlog.py

```diff
@@ -45,6 +50,14 @@ def test_fail():
         "$report_type": "SessionFinish",
     }
 
+    warning = json_objs.pop(12)
```

Comment:
Instead of popping the 12th item, can we please instead do an explicit loop? This will be easier to maintain.

```suggestion
    [warning] = [x for x in json_objs if x["$report_type"] == "warning-recorded"]
```

Reply:
agreed, though it should probably be something like
```python
    categorized = defaultdict(list)
    for x in json_objs:
        categorized[x["$report_type"] == "warning-recorded"].append(x)
    warning = categorized[True]
    json_objs = categorized[False]
```
since the code below is unable to properly check the results if the warning is still included